bundles/framework/layerlist/view/LayerCollapse/StateHandler.js

```diff
@@ -0,0 +1,106 @@
+export class StateHandler {
+    constructor () {
+        this.listeners = [];
+        this.selectedLayerIds = [];
+        this.groups = [];
+        this.filterKeyword = null;
+        this.openGroupTitles = null;
+        this.filtered = null;
+        this.map = Oskari.getSandbox().getMap();
+    }
+    getState () {
```

Comment:
Ability to get the internal state should not be needed. We could make it private and only call in `notify(). Storybooks etc. can still use the provate function "illegally", if needed.